Add support for content range requests when getting blobs (#537)
opencontainers/distribution-spec: spec.md

If the blob is not found in the registry, the response code MUST be `404 Not Found`.

A GET request may also include a `Range` request header to download part of a
blob. The response code can either be `216 (Partial Content)` or `416 (Range

> Not Satisfiable)` in case of an invalid range. If the registry doesn't support
> range requests, it can respond with `Accept-Ranges: none`.

Reviewer: ```suggestion
Not Satisfiable)` in case of an invalid range. A registry MAY ignore the `Range` header.
```

Author: An older registry unaware of this change may ignore the header, but a newer registry may want to send a response back to discourage clients from send the Range request.

If the blob is not found in the registry, the response code MUST be `404 Not Found`.

> A GET request MAY also include a `Range` request header to download part of a
> blob in accordance with [RFC 9110](https://datatracker.ietf.org/doc/html/rfc9110#name-range-requests).

Reviewer 2: I reworded it to focus on the registry since "a request MAY include" could imply "a registry MUST support" which I don't think is the goal.

Also a couple nits:

- One sentence per line for easier PRs (remove the linefeed here).
- I think we should standardize on a single RFC host. It looks like most of the links here use https://www.rfc-editor.org/.

```suggestion
A registry SHOULD support the `Range` request header in accordance with [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-range-requests).
```
